cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/AbstractProgramController.java

```diff
@@ -181,7 +181,9 @@ public void run() {
    * Children call this method to signal the program is completed.
    */
   protected void complete() {
-    if (!state.compareAndSet(State.ALIVE, State.COMPLETED)) {
+    if (!state.compareAndSet(State.STARTING, State.COMPLETED)
+      && !state.compareAndSet(State.ALIVE, State.COMPLETED)
+      && !state.compareAndSet(State.SUSPENDED, State.COMPLETED)) {
       LOG.debug("Cannot transit to COMPLETED state from {} state: {} {}", state.get(), name);
```

Comment:
Should we change this log message to a WARN or INFO since this indicates things are not as we expected. How frequently do we expect this to get logged in a typical system that is running fine?